docs/matlab_differences.rst

```diff
@@ -0,0 +1,68 @@
+Deliberate Differences from the Original PREP
+=============================================
+
+Although PyPREP aims to be a faithful reimplementaion of the original MATLAB
+version of PREP, there are a few places where PyPREP has deliberately chosen
+to use different defaults than the MATLAB PREP.
+
+To override these differerences, you can set the ``matlab_strict`` argument to
+:class:`pyprep.prep_pipeline.PrepPipeline`, :class:`pyprep.reference.Reference`,
+or :class:`pyprep.find_noisy_channels.NoisyChannels` as ``True`` to match the
+original PREP's internal math.
+
+Differences in RANSAC
+=====================
+
+During the "find-bad-by-RANSAC" step of noisy channel detection, PREP does the
+follwing steps to identify channels that aren't well-predicted by the signals
+of other channels:
+
+1) Generates a bunch of random subsets of currently-good channels from the data
+   (50 samples by default, each containing 25% of the total EEG channels in the
+   dataset).
+2) Uses the signals and spatial locations of those channels to predict what the
+   signals will be at the spatial locations of all the other channels, with each
+   random subset of channels generating a different prediction for each channel
+   (i.e., 50 predicted signals per channel by default).
+3) For each channel, calculates the median predicted signal from the full set of
+   predictions.
+4) Splits the full data into small non-overlapping windows (5 seconds by
+   default) and calculates the correlation between the median predicted signal
+   and the actual signal for each channel within each window.
+5) Compares the correlations for each channel against a threshold value (0.75
+   by default), flags all windows that fall below that threshold as 'bad', and
+   calculates the proportions of 'bad' windows for each channel.
+6) Flags all channels with an excessively high proportion of 'bad' windows
+   (minimum 0.4 by default) as being 'bad-by-RANSAC'.
+
+With that in mind, here are the areas where PyPREP's defaults deliberately
+differ from the original PREP implementation:
+
+Calculation of median estimated signal
+--------------------------------------
+
+In MATLAB PREP, the median signal in step 3 is calculated by sorting the
```

Comment:
would be nice to insert a link to their sourcecode here where this median calculation step takes place

Reply:
I tried including a permalink, but the URL for those commit lines was well over 120 characters so I just decided to add a code block with the relevant MATLAB code + a quick explanation of what it's doing. Let me know if you think that works!

Reply:
okay, that's fine - we have the permalink buried in our issues and PR conversations ... so if these spots in matprep code are ever needed again, we'll be able to dig them out. 👍